### PR TITLE
Switch to using OPA Constraint Framework engine.

### PR DESCRIPTION
**Pull request by briantkennedy**

### CODE REVIEW

File: pkg/gcv/configs/config.go [hunk 1]
```diff
@@ -206,9 +225,18 @@ func listFiles(dir string) ([]string, error) {
 
 // ListYAMLFiles returns a list of YAML files under a dir. Errors will be grpc errors.
 func ListYAMLFiles(dir string) ([]string, error) {
-	files, err := listFiles(dir)
-	if err != nil {
-		return nil, err
+	return ListYAMLFilesD([]string{dir})
+}
+
+// ListYAMLFiles returns a list of YAML files under a dir. Errors will be grpc errors.
```
Comment (t12g) [inline pkg/gcv/configs/config.go]: Let's update the godoc comment.
Is the "D" suffix a convention?

File: pkg/gcv/validator.go [hunk 2]
```diff
@@ -71,100 +75,51 @@ type Validator struct {
...
-			templates = append(templates, data)
-		case *configs.Constraint:
-			constraints = append(constraints, data)
-		default:
-			// Unexpected: CategorizeYAMLFile shouldn't return any types
-			return nil, nil, errors.Errorf("CategorizeYAMLFile returned unexpected data type when converting file %s", filePath)
-		}
-	}
-	return templates, constraints, nil
+	policyLibraryDir string
```
Comment (t12g) [inline pkg/gcv/validator.go]: can we delete this now?

**Comment by morgante** [comment]
@briantkennedy I believe merging this will require constraint template changes? We will need to coordinate between Terraform Validator, Forseti, and CFT Scorecard.

**Comment by briantkennedy** [comment]
@morgante yes, that's correct.  We will need to coordinate for making a versioned release so that we can roll forward the tooling and policy library when the time comes.

**Comment by briantkennedy** [comment]
I've updated this to have backward compatibility with the old template/constraint format.  After inspecting policy library, I was pleasantly surprised to find that the only policy that needs fixing is always_violates. @morgante any objections on merging?

**Pull request merged**